cmd/consumer/main.go

```diff
@@ -141,13 +140,10 @@ func appendToFile(filename, text string) error {
 }
 
 func (rp *RecordProcessor) Shutdown(checkpointer kcl.Checkpointer, reason string) error {
-	if reason == "TERMINATE" {
-		fmt.Fprintf(os.Stderr, "Was told to terminate, will attempt to checkpoint.\n")
-		rp.firehoseWriter.FlushAll()
-		rp.checkpoint(checkpointer, "", 0)
-	} else {
-		fmt.Fprintf(os.Stderr, "Shutting down due to failover. Reason: %s. Will not checkpoint.\n", reason)
-	}
+	fmt.Fprintf(os.Stderr, "Shutting down. Reason: %s. Attempting to flush messages and checkpoint.\n", reason)
+	rp.firehoseWriter.FlushAll()
+	time.Sleep(5 * time.Second) // Wait for messages to be flushed
```

Comment:
Another option would be to enhance the firehoseWriter so that it's `Flush()` method writes to a channel and listen for that channel so we know when the flush completes.